Task 1


In [4]:
import numpy as np

d1 = 3
d2 = 8

A = np.array([
    [d1 + 2, d2 + 1],
    [2 * d1, d2 + 2]
])

print("Matrix A:")
print(A)

print("\nShape of A:", A.shape)

det_A = np.linalg.det(A)
print("Determinant of A:", det_A)


print("Rank of A:", np.linalg.matrix_rank(A))


print("Eigenvalues of A:", np.linalg.eigvals(A))

# Inverse
if det_A != 0:
    print("Inverse of A:")
    print(np.linalg.inv(A))

Matrix A:
[[ 5  9]
 [ 6 10]]

Shape of A: (2, 2)
Determinant of A: -4.000000000000007
Rank of A: 2
Eigenvalues of A: [-0.26208735 15.26208735]
Inverse of A:
[[-2.5   2.25]
 [ 1.5  -1.25]]


Task 2


In [ ]:
# Create Matrix B by changing A[0,0]
B = A.copy()
B[0, 0] = B[0, 0] + 1

print("\nMatrix B:")
print(B)

print("\nShape of B:", B.shape)

det_B = np.linalg.det(B)
print("Determinant of B:", det_B)

print("Rank of B:", np.linalg.matrix_rank(B))

print("Eigenvalues of B:", np.linalg.eigvals(B))

# Inverse
if det_B != 0:
    print("Inverse of B:")
    print(np.linalg.inv(B))


Matrix B:
[[ 6  9]
 [ 6 10]]

Shape of B: (2, 2)
Determinant of B: 6.0
Rank of B: 2
Eigenvalues of B: [ 0.38422689 15.61577311]
Inverse of B:
[[ 1.66666667 -1.5       ]
 [-1.          1.        ]]


1.How did the determinant change and why?

The determinant of Matrix A was -4. After changing one element from 5 → 6, the determinant of Matrix B became 6. This happened because changing even one element in a matrix can affect the determinant calculation.

2.Did the rank change?

No, the rank did not change. Both Matrix A and Matrix B have a rank of 2. This is because both matrices are full-rank matrices and their determinants are not equal to zero.

3.How did the eigenvalues respond?

The eigenvalues of Matrix A were approximately -0.2621 and 15.2621. After changing one value, the eigenvalues of Matrix B became approximately 0.3842 and 15.6158. This shows that even a small change in a matrix value can change its eigenvalues.

4.Is B easier or harder to invert than A?

Both matrices are invertible because their determinants are not equal to zero. However, the absolute value of the determinant of Matrix B (|6|) is greater than that of Matrix A (|-4| = 4). Therefore, Matrix B is generally less close to being singular and may be slightly more numerically stable when calculating its inverse.